libraries/freertos_plus/standard/pkcs11/utest/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+project ("freertos_plus pkcs11 cmock unit test")
+cmake_minimum_required (VERSION 3.13)
+set(3rdparty_dir "${AFR_ROOT_DIR}/libraries/3rdparty")
+set(standard_dir "${AFR_ROOT_DIR}/libraries/freertos_plus/standard")
+set(kernel_dir "${AFR_ROOT_DIR}/freertos_kernel")
+
+set(preprocess_commands "-I ${3rdparty_dir}/pkcs11/ -D'CK_PTR=*' -D'NULL_PTR=0' \
+                         -D'CK_DEFINE_FUNCTION( returnType, name )=returnType name'\ 
+                         -D'CK_DECLARE_FUNCTION( returnType, name )=returnType name'\ 
+                         -D'CK_DECLARE_FUNCTION_POINTER( returnType, name )=returnType( CK_PTR name )'\
+                         -D'CK_CALLBACK_FUNCTION( returnType, name )=returnType( CK_PTR name )'")
+
+separate_arguments(unix_flags UNIX_COMMAND "${preprocess_commands}")
+
+# Mock library
+list(APPEND mock_list
+    "${kernel_dir}/include/portable.h"
+    "${3rdparty_dir}/pkcs11/pkcs11.h"
+    )
+
+list(APPEND preprocessed_mock_list
+    "${3rdparty_dir}/pkcs11/pkcs11.h"
+    )
+
+
+create_mock_list(pkcs11_mock "${mock_list}")
+preprocess_mock_list(pkcs11_mock ${preprocessed_mock_list} "${unix_flags}")
+
+
+target_compile_definitions(pkcs11_mock PUBLIC
+        portHAS_STACK_OVERFLOW_CHECKING=1
+        portUSING_MPU_WRAPPERS=1
+        MPU_WRAPPERS_INCLUDED_FROM_API_FILE
+        )
+
+add_dependencies(pkcs11_mock
+                 cmock
+                 unity
+                 pre_pkcs11_mock
+                 )
+
+# Real libraries
+add_library(pkcs11_real STATIC
+        ${standard_dir}/pkcs11/src/iot_pkcs11.c
+        )
+
+target_include_directories(pkcs11_real PUBLIC
+        ${3rdparty_dir}/pkcs11/
+        ${standard_dir}/pkcs11/include
+        ${CMAKE_CURRENT_BINARY_DIR}/mocks
+        )
+
+set_target_properties(pkcs11_real PROPERTIES
+        COMPILE_FLAGS "-Wall -fPIC -ggdb3 -Og \
```

Comment:
We should be able to run unit tests both with and without debugging.  Normally we set debugging flags in cmake using `CMAKE_C_FLAGS_DEBUG` like [this](https://github.com/aws/amazon-freertos/blob/master/vendors/pc/boards/linux/CMakeLists.txt#L3).  You might see if you can set these up in a similar way

Reply:
Actually since @gkwicker approved my commit today, we can just remove the switches on that line

Reply:
I have now removed these flags.